test/JDBC/README.md

```diff
@@ -1,5 +1,377 @@
+
+Connection attributes should be specified in pairs:
+```
+<connection attribute 1>=<value 1>     <connection attribute 2>=<value 2> ...
+```
+
+Currently the attributes that can specified are `user`, `password`, `database` and `currentSchema` (only for PG connection).
+
+**Example**
+```tsql
+-- tsql
+CREATE PROCEDURE tsql_interop_proc1
+AS
+UPDATE procTab1 SET c1=10 where c2='b';
+INSERT INTO procTab1 values(1,'a');
+INSERT INTO procTab1 values(2,'b');
+EXEC psql_interop_proc2;
+GO
+
+-- psql     currentSchema=master_dbo,public
+CREATE PROCEDURE psql_interop_proc1()
+AS
+$$
+BEGIN
+    CREATE TABLE procTab1(c1 int);
+    INSERT INTO procTab1 values (5);
+    ALTER TABLE procTab1 ADD c2 char;
+    CALL tsql_interop_proc1();
+END
+$$ LANGUAGE PLPGSQL;
+GO
+
+CREATE PROCEDURE psql_interop_proc2()
+AS
+$$
+BEGIN
+    INSERT INTO procTab1(c1,c2) values (3,'c');
+    UPDATE procTab1 SET c1=10 where c2='b';
+    INSERT INTO procTab1(c1,c2) values (4,'d');
+    DELETE FROM procTab1 where c2='a';
+END
+$$ LANGUAGE PLPGSQL;
+GO
+
+-- tsql
+EXEC psql_interop_proc1
+GO
+SELECT * FROM procTab1 order by c1
+GO
+
+-- psql     currentSchema=master_dbo,public
+CALL tsql_interop_proc1()
+GO
+SELECT DISTINCT c1 FROM procTab1
+GO
+```
+
+Input file type: `.mix`
+
+---
+
+### **IMPORTANT**
+- If you want to execute a SQL Batch in `.txt` input files, you will need to specify the batch in a single line without the `GO` batch separator. This is needed because for `.txt` files, the test framework treates every line as a standalone statement/command that can be executed against the server.
+- You CANNOT club functionalities of different file types. For example, you cannot execute prep-exec statements (functionality of `.txt` input file) in a `.mix` file. 
+
+## Adding tests
+The test framework consumes `.sql`, `.txt` and `.mix` files as input (discussed above) and uses them to generate the output (.out) files.
+
+1. Add the input test file in the `test/JDBC/input` directory. You can also create subdirectories and test files inside those
+2. Run the test framework to generate the output file (you may want to edit the schedule file to only the run the tests that you have added instead of running all the tests). The test will fail because it does not have a corresponding expected output file yet.
+3. Check the `test/JDBC/output` directory for the generated output file. The output file (.out) will be of the format:
+    ```
+    test/JDBC/output/<your_test_filename>.out
+    ```
+4. If the output file looks correct to you, move it to the `test/JDBC/expected` directory:
+    ```
+    mv test/JDBC/output/<your_test_filename>.out test/JDBC/expected
+    ```
+5. Re-run the test framework and ensure that the newly added test now passes
+
+## Reading the console output and diff
+If all the tests PASS, `TESTS FAILED` will be zero and you will be greeted with a `BUILD SUCCESS` message
```

Comment:
Punctuation:

If all the tests PASS, `TESTS FAILED` will be zero and you will be greeted with a `BUILD SUCCESS` message: